src/fromager/commands/bootstrap.py

```diff
@@ -72,3 +73,22 @@ def bootstrap(
                 f"removing prebuilt wheel {prebuilt_wheel.name} from download cache"
             )
             filename.unlink()
+
+    constraints_filename = wkctx.work_dir / "constraints.txt"
+    logger.info(f"writing installation dependencies to {constraints_filename}")
+    with open(wkctx.work_dir / "constraints.txt", "w") as f:
+        for name, version in sorted(
+            set(_installation_dependencies(wkctx.all_edges, ""))
+        ):
+            f.write(f"{name}=={version}\n")
+
+
+def _installation_dependencies(
+    all_edges: dict[NormalizedName | str, tuple[str, NormalizedName, str]],
```

Comment:
```suggestion
    all_edges: dict[NormalizedName, list[tuple[str, NormalizedName, Version]]],
```
Syncing up the type with `all_edges` defined in context.py. It might be helpful to define a reusable type alias for the edges/graph in this case

Reply:
I kept "str" for the key type because one key is `""` and that's not really a NormalizedName. Maybe that's not necessary?

You're right about the list, I'll make that update.

Reply:
Done